src/System.Net.Primitives/src/System/Net/CredentialCache.cs

```diff
@@ -0,0 +1,523 @@
+
+            if (comparand == null)
+            {
+                // This covers also the compared == null case
+                return false;
+            }
+
+            bool equals =
+                (string.Compare(AuthenticationType, comparedCredentialKey.AuthenticationType, StringComparison.OrdinalIgnoreCase) == 0) &&
+                (string.Compare(Host, comparedCredentialKey.Host, StringComparison.OrdinalIgnoreCase) == 0) &&
+                Port == comparedCredentialKey.Port;
+
+            GlobalLog.Print("CredentialKey::Equals(" + ToString() + ", " + comparedCredentialKey.ToString() + ") returns " + equals.ToString());
+
+            return equals;
+        }
+
+        public override string ToString()
+        {
+            return "[" + Host.Length.ToString(NumberFormatInfo.InvariantInfo) + "]:" + Host + ":" + Port.ToString(NumberFormatInfo.InvariantInfo) + ":" + Logging.ObjectToString(AuthenticationType);
+        }
+    }
+
+    internal class CredentialKey
+    {
+        internal Uri UriPrefix;
+        internal int UriPrefixLength = -1;
+        internal string AuthenticationType;
+
+        internal CredentialKey(Uri uriPrefix, string authenticationType)
+        {
+            UriPrefix = uriPrefix;
+            UriPrefixLength = UriPrefix.ToString().Length;
+            AuthenticationType = authenticationType;
+        }
+
+        internal bool Match(Uri uri, string authenticationType)
+        {
+            if (uri == null || authenticationType == null)
+            {
+                return false;
+            }
+
+            // If the protocols don't match, this credential is not applicable for the given Uri.
+            if (string.Compare(authenticationType, AuthenticationType, StringComparison.OrdinalIgnoreCase) != 0)
+            {
+                return false;
+            }
+
+            GlobalLog.Print("CredentialKey::Match(" + UriPrefix.ToString() + " & " + uri.ToString() + ")");
+
+            return IsPrefix(uri, UriPrefix);
+        }
+
+        // IsPrefix (Uri)
+        //
+        // Determines whether <prefixUri> is a prefix of this URI. A prefix
+        // match is defined as:
+        //
+        //     scheme match
+        //     + host match
+        //     + port match, if any
+        //     + <prefix> path is a prefix of <URI> path, if any
+        //
+        // Returns:
+        // True if <prefixUri> is a prefix of this URI
+        internal bool IsPrefix(Uri uri, Uri prefixUri)
+        {
+            if (prefixUri.Scheme != uri.Scheme || prefixUri.Host != uri.Host || prefixUri.Port != uri.Port)
+            {
+                return false;
+            }
+
+            int prefixLen = prefixUri.AbsolutePath.LastIndexOf('/');
+            if (prefixLen > uri.AbsolutePath.LastIndexOf('/'))
+            {
+                return false;
+            }
+
+            return String.Compare(uri.AbsolutePath, 0, prefixUri.AbsolutePath, 0, prefixLen, StringComparison.OrdinalIgnoreCase) == 0;
```

Comment:
string.Equals? I'll stop commenting on these.  If you choose to fix, I'd suggest just doing a search for all of them.